precompiles/staking/method_claim_rewards.go

```diff
@@ -0,0 +1,151 @@
+package staking
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+
+	"github.com/zeta-chain/node/cmd/zetacored/config"
+	"github.com/zeta-chain/node/precompiles/bank"
+	precompiletypes "github.com/zeta-chain/node/precompiles/types"
+	fungibletypes "github.com/zeta-chain/node/x/fungible/types"
+)
+
+// claimRewards claims all the rewards for a delegator from a validator.
+// As F1 Cosmos distribution scheme implements an all or nothing withdrawal, the precompile will
+// withdraw all the rewards for the delegator, filter ZRC20 and unlock them to the delegator EVM address.
+func (c *Contract) claimRewards(
+	ctx sdk.Context,
+	evm *vm.EVM,
+	_ *vm.Contract,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	if len(args) != 2 {
+		return nil, &precompiletypes.ErrInvalidNumberOfArgs{
+			Got:    len(args),
+			Expect: 2,
+		}
+	}
+
```

Comment:
We removed `originIsDelegator = evm.Origin == delegatorAddr` but don't we need to keep this check?
```
callerIsDelegator = contract.CallerAddress == delegatorAddr
```

Reply:
In this case contract.CallerAddress is the immediate caller, that would be the dApp. The tx.Origin would be the initiator of the tx.

Example running our e2e through contract:
DEBUG: ERROR
DEBUG: contract.CallerAddress 0xc27487f904180c26B30C1E983261f9bA4A31f5D7
DEBUG: delegatorAddr 0xb552FFAb2500258C1A705Ba4Fe77A333275AFE45

The delegatorAddr is the e2e user and the CallerAddress. The other address is the deployed dApp.

Reply:
Because of the above I believe it makes sense to check that delegator address is the origin or the callerAddress.